server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -138,26 +148,95 @@ protected void clusterManagerOperation(
+                                IndexShardStats shard = indicesStatsResponse.getIndex(sourceIndex).getIndexShards().get(i);
+                                return shard == null ? null : shard.getPrimary().getDocs();
+                            }, indicesStatsResponse.getPrimaries().store, sourceIndex, targetIndex);
+
+                            if (verifyDocCountOnReplicationGroup(indicesStatsResponse, sourceIndex) == false) {
+                                throw new IllegalStateException(
+                                    " For index ["
+                                        + sourceIndex
+                                        + "] replica shards haven't caught up with primary, please retry after sometime."
+                                );
+                            }
+
+                            createIndexService.createIndex(
+                                updateRequest,
+                                ActionListener.map(
+                                    delegatedIndicesStatsListener,
+                                    response -> new ResizeResponse(
+                                        response.isAcknowledged(),
+                                        response.isShardsAcknowledged(),
+                                        updateRequest.index()
+                                    )
+                                )
+                            );
+                        }));
+                }));
+        } else {
+            client.admin()
+                .indices()
+                .prepareStats(sourceIndex)
+                .clear()
+                .setDocs(true)
+                .setStore(true)
+                .execute(ActionListener.delegateFailure(listener, (delegatedListener, indicesStatsResponse) -> {
+                    CreateIndexClusterStateUpdateRequest updateRequest = prepareCreateIndexRequest(resizeRequest, state, i -> {
+                        IndexShardStats shard = indicesStatsResponse.getIndex(sourceIndex).getIndexShards().get(i);
+                        return shard == null ? null : shard.getPrimary().getDocs();
+                    }, indicesStatsResponse.getPrimaries().store, sourceIndex, targetIndex);
+                    createIndexService.createIndex(
+                        updateRequest,
+                        ActionListener.map(
+                            delegatedListener,
+                            response -> new ResizeResponse(
+                                response.isAcknowledged(),
+                                response.isShardsAcknowledged(),
+                                updateRequest.index()
+                            )
+                        )
+                    );
+                }));
+        }
+
+    }
+
+    private boolean verifyDocCountOnReplicationGroup(IndicesStatsResponse indicesStatsResponse, String index) {
```

Comment:
We shouldn't be using doc count here as a reliable measurement for if the replica is current, this will not be accurate post merges.  We could use bytes behind that will be introduced with https://github.com/opensearch-project/OpenSearch/issues/9417.

Reply:
Another alternative is seqNo stats.  However, these aren't as reliable with segRep because we only update the processed after a new commit has been made on the primary.  So the replica could be caught up and not reflect that in its seqNo stats.

Reply:
Yes I have updated that in latest commit and added check on bytes behind